bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -825,6 +890,22 @@ public void remove(Object element) {
 	public void reveal(Object element) {
```

Comment:
I'm not sure if this is used for the link with editor? But link with editor seems to work oddly. Sometimes it will result in selecting the expandable node, sometimes it will result in selecting the parent of the expandable node, sometimes nothing is selected. I think different views behave differently for it (e.g. Package Explorer vs Project Explorer).

Reply:
yes. package exlorer has fail safe client side mechanism. If it fails to select an element it tris to select parent.
But Project exlorer don't have that. May be need to add same here as well ?
In Package explorer selection is purely based on available expandable nodes direct children at that moment. If we on't find element no selection is made.

Reply:
> May be need to add same here as well ?

I think this can be postponed for later polishing of the patch, the difference is not important.